### PR TITLE
Make OpenZip less strict, add OpenZipWithName

### LINKED ISSUE #14: ZIPReader should make less assumptions on file name
(idea originally by @tdilo)

At the moment `OpenZip` assumes that the shapefile lies in the root of the archived filetree and has the same basename as the archive file.

The idea is to lift the restriction by scanning through the archive and using the relative path and  basename of the first found shapefile.

### PR DESCRIPTION
**Pull request by fawick**
Closes #14.

### CODE REVIEW

File: zipreader.go [hunk 1]
```diff
@@ -27,25 +26,69 @@ func openFromZIP(z *zip.ReadCloser, name string) (io.ReadCloser, error) {
...
+		return nil, err
+	}
+	// dbf is optional, so no error checking here
+	dbf, _ := openFromZIP(zr.z, prefix+".dbf")
+	zr.sr = SequentialReaderFromExt(shp, dbf)
+	return zr, nil
+}
+
+// OpenZipWithName opens a shape file that is contained in a ZIP achive. The
+// parameter name name of the shape file.
```
Comment (tdilo) [inline zipreader.go]: "name is the name"?

File: zipreader.go [hunk 2]
```diff
@@ -27,25 +26,69 @@ func openFromZIP(z *zip.ReadCloser, name string) (io.ReadCloser, error) {
...
+// It must be a relative path: it must not start with a drive
```
Comment (tdilo) [inline zipreader.go]: This is ambiguous. Does "it" refer to the name of the shapefile or the zip file path?
IMO the wording "relative path" is clear enough on its own and does not require further explanation.

File: zipreader.go (hunk 1 above)
Comment (fawick) [reply zipreader.go]: Done.

File: zipreader.go (hunk 2 above)
Comment (fawick) [reply zipreader.go]: Done.

**Comment by fawick** [comment]
@tdilo Thx for the review, pls take another look.

File: zipreader.go [hunk 3]
```diff
@@ -27,25 +26,81 @@ func openFromZIP(z *zip.ReadCloser, name string) (io.ReadCloser, error) {
...
+	}
+	withoutExt := strings.TrimSuffix(shapeFiles[0].Name, ".shp")
+	// dbf is optional, so no error checking here
+	dbf, _ := openFromZIP(zr.z, withoutExt+".dbf")
+	zr.sr = SequentialReaderFromExt(shp, dbf)
+	return zr, nil
+}
+
+// ShapesInZip returns a string-slice with the names (i.e. relatives paths in
+// achive file tree) of all shapes that are in the ZIP archive at zipFilePath.
```
Comment (tdilo) [inline zipreader.go]: archive

**Pull request merged**